yellow-paper/docs/public-vm/state.md

```diff
@@ -0,0 +1,107 @@
+
+Aztec state is exposed to the AVM with access limitations that are stricter in some cases than they are for private execution. The relevant state databases are listed below alongside their underlying tree types (when applicable) and the access limitations imposed for the AVM.
+
+| State DB             | State Tree          | Tree Type                                       | Access                                                |
+| ---                  | ---                 | ---                                             | ---                                                   |
+| `noteHashDB`         | `noteHashTree`      | Append-only, non-indexed merkle tree            | appends, historic root                                |
+| `nullifierDB`        | `nullifierTree`     | Indexed merkle tree                             | membership-checks, appends, historic root             |
+| `l1ToL2MessageDB`    | `l1ToL2MessageTree` | Indexed merkle tree                             | membership-checks, leaf-preimage-reads, historic root |
+| `publicDataDB`       | `publicDataTree`    | Indexed merkle tree                             | membership-checks, reads, writes, historic root       |
+| `archiveDB`          | `archiveTree`       | Append-only, non-indexed merkle tree            | membership-checks, historic root                      |
+| `contractClassDB`    | -                   | -                                               | read, write                                           |
+| `contractInstanceDB` | -                   | -                                               | read, write                                           |
+
+> As described in ["Contract Deployment"](../contract-deployment), contracts are not stored in any single tree. A [contract class](../contract-deployment/classes) is [represented](../contract-deployment/classes#registration) as an unencrypted log containing the `ContractClass` structure (which contains the bytecode) and a nullifier representing the class identifier. The `contractClassDB` stores contract classes indexed by class identifier. A [contract instance](../contract-deployment/instances) is [represented](../contract-deployment/classes#registration) as an unencrypted log containing the `ContractInstance` structure and a nullifier representing the contract address. The `contractInstanceDB` stores contract instances indexed by contract address.
+
+### AVM World State
+
+The AVM does not directly modify Aztec state. Instead, it stages modifications to be applied later pending successful execution. As part of each execution context, the AVM maintains **world state** which is a representation of Aztec state that includes _staged_ modifications.
```

Comment:
I think not implementing changes immediately is more of an optimisation detail vs yp content

Reply:
Hmm, maybe.... I feel like it's important to note that the AVM doesn't _own_ Aztec state. And even though it might be a bit in the weeds, I feel like it's helpful for the reader to understand that each context maintains its own staged modifications that may or may not be accepted by the caller. And for the TX itself, "accepting" means committing to Aztec state.

Reply:
Decided to leave this in there for now.... Can still be convinced to remove later.